src/container/media.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect, useState } from 'react';
+import { Breakpoint, BREAKPOINTS_DESCENDING, matchBreakpointMapping } from '../internal/breakpoints';
+import { ContainerProps, MediaDefinition } from './interfaces';
+import { useContainerBreakpoints } from '../internal/hooks/container-queries';
+
+type MediaProps = keyof ContainerProps.Media;
+
+const MEDIA_PROPS: MediaProps[] = ['position', 'width', 'height', 'content'];
+
+const hasDefinedBreakpoint = (obj: any) => {
```

Comment:
I think we can do better than `any`. Maybe something like `obj: { breakpoint: Breakpoint }` ? Or a generic type